src/app/shared/object-list/my-dspace-result-list-element/item-list-preview/item-list-preview.component.ts

```diff
@@ -0,0 +1,60 @@
+import { Component, Input } from '@angular/core';
+
+import { Item } from '../../../../core/shared/item.model';
+import { fadeInOut } from '../../../animations/fade';
+import { MyDspaceItemStatusType } from '../../../object-collection/shared/mydspace-item-status/my-dspace-item-status-type';
+import { Metadata } from '../../../../core/shared/metadata.utils';
+import { MyDSpaceResult } from '../../../../+my-dspace-page/my-dspace-result.model';
+
+/**
+ * This component show metadata for the given item object in the list view.
+ */
+@Component({
+  selector: 'ds-item-list-preview',
+  styleUrls: ['item-list-preview.component.scss'],
+  templateUrl: 'item-list-preview.component.html',
+  animations: [fadeInOut]
+})
+export class ItemListPreviewComponent {
+
+  /**
+   * The item to display
+   */
+  @Input() item: Item;
+
+  /**
+   * The mydspace result object
+   */
+  @Input() object: MyDSpaceResult<any>;
+
+  /**
+   * Represent item's status
+   */
+  @Input() status: MyDspaceItemStatusType;
+
+  /**
+   * A boolean representing if to show submitter information
+   */
+  @Input() showSubmitter = false;
+
+  /**
+   * Gets all matching metadata string values from hitHighlights or dso metadata, preferring hitHighlights.
+   *
+   * @param {string|string[]} keyOrKeys The metadata key(s) in scope. Wildcards are supported; see [[Metadata]].
+   * @returns {string[]} the matching string values or an empty array.
+   */
+  allMetadataValues(keyOrKeys: string | string[]): string[] {
```

Comment:
these allMetadataValues and firstMetadataValue methods are used in a couple of places, they can probably be refactored

Reply:
fixed